x-pack/plugins/security/server/routes/session_management/info.ts

```diff
@@ -11,19 +11,23 @@ import type { SessionInfo } from '../../../common/types';
 /**
  * Defines routes required for the session info.
  */
-export function defineSessionInfoRoutes({ router, getSession }: RouteDefinitionParams) {
+export function defineSessionInfoRoutes({ router, getSession, config }: RouteDefinitionParams) {
   router.get(
     { path: '/internal/security/session', validate: false },
     async (_context, request, response) => {
       const sessionValue = await getSession().get(request);
       if (sessionValue) {
+        const expirationTime =
+          sessionValue.idleTimeoutExpiration && sessionValue.lifespanExpiration
+            ? Math.min(sessionValue.idleTimeoutExpiration, sessionValue.lifespanExpiration)
+            : sessionValue.idleTimeoutExpiration || sessionValue.lifespanExpiration;
+
         return response.ok({
           body: {
-            // We can't rely on the client's system clock, so in addition to returning expiration timestamps, we also return
-            // the current server time -- that way the client can calculate the relative time to expiration.
-            now: Date.now(),
-            idleTimeoutExpiration: sessionValue.idleTimeoutExpiration,
-            lifespanExpiration: sessionValue.lifespanExpiration,
+            expiresInMs: expirationTime ? expirationTime - Date.now() : null,
+            canBeExtended:
+              sessionValue.idleTimeoutExpiration !== null &&
+              expirationTime !== sessionValue.lifespanExpiration,
             provider: sessionValue.provider,
           } as SessionInfo,
         });
```

Comment:
I like your approach here, it makes more sense than sending three dates to the client.

I did notice one bug that users can run into though -- if `idleTimeoutExpiration` and `lifespanExpiration` are both non-null and they are less than 5 minutes apart (the amount of time that the warning toast is shown to the user), you can run into this situation:

![session](https://user-images.githubusercontent.com/5295965/116723861-eb79da80-a9ad-11eb-85a9-50cede58f118.gif)

Technically, was the user able to extend his session? Yes... but it would be frustrating to run into this.

Before, when we were sending the `lifespanExpiration` to the client, we were able to account for this and handle it accordingly (IIRC we just showed the "You will need to log in again" version of the toast earlier).

One way to prevent this behavior is:
1. Move `WARNING_MS` to `x-pack/plugins/security/common/constants.ts` (and maybe rename it to `SESSION_EXPIRATION_WARNING_MS`?)
2. Then change this to use something like:
   ```ts
   const canBeExtended =
     sessionValue.idleTimeoutExpiration !== null &&
     expirationTime !== null &&
     (sessionValue.lifespanExpiration === null ||
       expirationTime + SESSION_EXPIRATION_WARNING_MS < sessionValue.lifespanExpiration);
   ```

Of course that wouldn't actually prevent the session from being extended, though we could modify `x-pack/plugins/security/server/session_management/session.ts` to take that into account too? 🤷‍♂️